src/Dialect/ONNX/MLIRDialectBuilder.hpp.inc

```diff
@@ -0,0 +1,135 @@
+//===- MLIRDialectBuilder.inc - Support migration to MLIR without EDSC --===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
```

Comment:
Did this code actually come from llvm?

I'm curious why this is a `.inc` file rather than just a `.hpp` or `.cpp` file.

Reply:
I have it as an include file so as to not "pollute" the header file with the implementation, which is needed as this is a template.

Reply:
The Dialect concept is lifted from MLIR, but modified here to use a base class, additional methods. So its a derivative work.